src/openrct2/world/sprite.c

```diff
@@ -267,6 +267,7 @@ void sprite_clear_all_unused()
         sprite->previous = previousSpriteIndex;
         sprite->linked_list_type_offset = SPRITE_LIST_NULL * 2;
         sprite->sprite_index = spriteIndex;
+        sprite->next_in_quadrant = (sprite->next_in_quadrant == 0) ? SPRITE_INDEX_NULL : sprite->next_in_quadrant;
```

Comment:
A comment would be nice, also this is self-assignment. Please change it to this form `if (sprite->next_in_quadrant == 0) { sprite->next_in_quadrant = SPRITE_INDEX_NULL; }`